src/JwplatformClient.php

```diff
@@ -0,0 +1,327 @@
+
+    public function list($site_id, $channel_id, $query_params = null) {
+        return $this->_client->request(
+            "GET",
+            "/v2/sites/" . $site_id . "/channels/" . $channel_id . "/events/",
+            null,
+            [],
+            $query_params
+        );
+    }
+
+    public function get($site_id, $channel_id, $resource_id, $query_params = null) {
+        return $this->_client->request(
+            "GET",
+            "/v2/sites/" . $site_id . "/channels/" . $channel_id . "/events/" . $resource_id . "/",
+            null,
+            [],
+            $query_params
+        );
+    }
+
+    public function request_master($site_id, $channel_id, $resource_id, $query_params = null) {
+        return $this->_client->request(
+            "PUT",
+            "/v2/sites/" . $site_id . "/channels/" . $channel_id . "/events/" . $resource_id . "/request_master/",
+            null,
+            [],
+            $query_params
+        );
+    }
+
+    public function clip($site_id, $channel_id, $resource_id, $body = null, $query_params = null) {
+        return $this->_client->request(
+            "PUT",
+            "/v2/sites/" . $site_id . "/channels/" . $channel_id . "/events/" . $resource_id . "/clip/",
+            $body,
+            [],
+            $query_params
+        );
+    }
+}
+
+class _MediaClient extends _SiteResourceClient {
+
+    protected $_resource_name = "media";
+
+    public function reupload($site_id, $resource_id, $body, $query_params = null) {
+        return $this->_client->request(
+            "PUT",
+            $this->_path_format($this->_singular_path, $site_id, $this->_resource_name, $resource_id) . "reupload/",
+            $body,
+            [],
+            $query_params
+        );
+    }
+
+    /**
+     * Upload a media file
+     *
+     * @param string $file_path Local path to the file to be uploaded
+     * @param string $upload_url Destination URL from the media create response
+     * @return string
+     * @throws Exception
+     */
+    public function upload($file_path, $upload_url) {
+        $post_data = file_get_contents($file_path);
+        $expected_hash = @md5_file($file_path);
+
+        $curl = curl_init();
+        curl_setopt($curl, CURLOPT_URL, $upload_url);
+        curl_setopt($curl, CURLOPT_CUSTOMREQUEST, 'PUT');
+        curl_setopt($curl, CURLOPT_POSTFIELDS, $post_data);
+        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($curl, CURLOPT_HEADER, true);
+        curl_setopt($curl, CURLINFO_HEADER_OUT, true);
+        curl_setopt($curl, CURLOPT_HTTPHEADER, array(
+            'Content-Type:',
+        ));
+        $response = curl_exec($curl);
+        $curl_info = curl_getinfo($curl);
+        curl_close($curl);
+
+        $header_size = $curl_info['header_size'];
+        $header = substr($response, 0, $header_size);
+        preg_match("/ETag: \"(.*?)\"\r\n/", $header, $matches);
```

Comment:
Do we implement this behavior in other clients? Don't remember seeing this in the python v2 client

Reply:
Not every client but it's a nice to have feature that was added to the python one: https://github.com/jwplayer/jwplatform-py/blob/e9db16462daf9931ac0e9f616c58ef529eb69aa1/jwplatform/upload.py#L227